clang-tools-extra/clang-tidy/modernize/ReturnBracedInitListCheck.cpp

```diff
@@ -17,11 +17,21 @@ using namespace clang::ast_matchers;
 namespace clang::tidy::modernize {
 
 void ReturnBracedInitListCheck::registerMatchers(MatchFinder *Finder) {
-  // Skip list initialization and constructors with an initializer list.
+  auto SemanticallyDifferentContainer = allOf(
+      argumentCountIs(2), hasArgument(0, hasType(isInteger())),
```

Comment:
I'm not 100% convinced that isInteger allone will be sufficient, better would be something like:
`hasType(qualType(hasCanonicalType(isInteger())))`